homework/list.js

```diff
@@ -52,28 +44,41 @@
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    this._wrapper.appendChild(ul);
+  },
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
+  fetchList() {
+    return new Promise(function(resolve, reject) {
+		var xhr = new XMLHttpRequest();
+        xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+        xhr.responseType = 'json';
+        xhr.onreadystatechange = (function(e) {
+          // Watch out: we have a mysterious unknown 'this'.
+          if (this.readyState === 4 && this.status === 200) {
+            var listData = this.response;
+            // The flow ends here.
+            afterFetch(listData);
```

Comment:
**WRONG**: after wrapping a Promise, you should get rid of the old callback style. This shows you failed to understand what the Promise is, and how to use that. **Please check examples as this to correct your patch**:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Examples

And you even didn't check if `afterFetch` exists. It should **never happen** if you ever write and run the code carefully.